### PR TITLE
Replicas > 1 is not allowed with sqlite driver

### PR DESCRIPTION
**Pull request by kfox1111**
This pr enforces replicas of the spire server can not be more then one when the sqlite driver is used.

fixes: https://github.com/spiffe/helm-charts/issues/268

### CODE REVIEW

**Comment by kfox1111** [comment]
> Might be better to merge this after the database pr as youll have to change it to check for database type.

Was thinking this is standalone enough it should review pretty fast and we can get it in, then I'd extend it in the db pr real quick to work properly with it?

File: charts/spire/charts/spire-server/templates/hpa.yaml [hunk 1]
```diff
@@ -12,6 +12,9 @@ spec:
     kind: Deployment
     name: {{ include "spire-server.fullname" . }}
   minReplicas: {{ .Values.autoscaling.minReplicas }}
+  {{- if gt (int .Values.autoscaling.maxReplicas) 1 }}
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/templates/hpa.yaml:15]: If external db merges first, then this becomes something like:
{{- if and (eq .Values.dataStore.sql.databaseType "sqlite3") (gt (int .Values.autoscaling.maxReplicas) 1) }}

Comment (faisal-memon) [reply charts/spire/charts/spire-server/templates/hpa.yaml:15]: Yes, lets merge the external db first and then this change.

Comment (kfox1111) [reply charts/spire/charts/spire-server/templates/hpa.yaml]: Reordered to come after external db change.

File: charts/spire/charts/spire-server/templates/hpa.yaml [hunk 2]
```diff
@@ -12,6 +12,9 @@ spec:
     kind: Deployment
     name: {{ include "spire-server.fullname" . }}
   minReplicas: {{ .Values.autoscaling.minReplicas }}
+  {{- if and (eq .Values.dataStore.sql.databaseType "sqlite3") .Values.autoscaling.enabled (gt (int .Values.autoscaling.maxReplicas) 1) }}
+  {{- fail "sqlite3 database only supports one replica. You can't use autoscaling.maxReplicas > 1" }}
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/templates/hpa.yaml:16]:
```suggestion
  {{- fail "When running with sqlite3 database, you can't scale up to more then one instance. 'autoscaling.maxReplicas' MUST be 1" }}
```

File: charts/spire/charts/spire-server/templates/statefulset.yaml [hunk 3]
```diff
@@ -11,6 +11,9 @@ metadata:
     {{- include "spire-server.labels" . | nindent 4 }}
 spec:
   {{- if not .Values.autoscaling.enabled }}
+  {{- if and (eq .Values.dataStore.sql.databaseType "sqlite3") (gt (int .Values.replicaCount) 1) }}
+  {{- fail "sqlite3 database only supports one replica." }}
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/templates/statefulset.yaml:15]:
```suggestion
  {{- fail "When running with sqlite3 database, you can't scale up to more then one instance. 'replicaCount' MUST be 1" }}
```

**Review by marcofranssen — COMMENTED** [review]
Small suggestion on the failure message to have it aligned in style accross both scenarios.

I'm also good merging this PR first. It doesn't break anything and is a nice improvement to prevent misusing the chart.

👍

**Pull request merged**